pkg/plugins/manager/installer.go

```diff
@@ -24,23 +25,26 @@ type PluginInstaller struct {
 	pluginRegistry       registry.Service
 	pluginLoader         loader.Service
 	log                  log.Logger
+	serviceRegistry      auth.ExternalServiceRegistry
```

Comment:
Looks like we can use this logic as part of the plugin loader setup rather than directly in the uninstaller - we can add it as part of the termination steps [here](https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/pipeline/pipeline.go#L78), like we've done with the registration step [here](https://github.com/grafana/grafana/blob/main/pkg/services/pluginsintegration/pipeline/pipeline.go#L66).

Reply:
We talked about this, the problem is that we don't want to remove service registrations when shutting down (so we don't have to recreate it on start up). The service registry on start up will be a no-op if the service already exists.